api/v1alpha1/common_types.go

```diff
@@ -47,6 +47,24 @@ type ControlPlaneVirtualIPSpec struct {
 	// +kubebuilder:validation:Enum=KubeVIP
 	// +kubebuilder:default=KubeVIP
 	Provider string `json:"provider,omitempty"`
+
+	// Configuration for the chosen control-plane virtual IP provider.
+	// +kubebuilder:validation:Optional
+	Configuration *ControlPlaneVirtualIPConfiguration `json:"configuration,omitempty"`
+}
+
+type ControlPlaneVirtualIPConfiguration struct {
+	// The virtual IP on which the API server is serving.
+	// If left empty, the value from controlPlaneEndpoint.host will be used.
+	// +kubebuilder:validation:Optional
```

Comment:
```suggestion
	// +kubebuilder:validation:Optional
	// +kubebuilder:validation:Format=ipv4
```

Reply:
Could also be ipv6...

Reply:
Let's add ipv4 format as @supershal suggests for now, and once we upgrade to CAPI 1.9 we can start to use `oneOf` schemas to support both ipv4 and ipv6 here.